lang/compiler.go

```diff
@@ -391,6 +387,25 @@ func (p *AstProgram) getSpec(ctx context.Context, runtime flux.Runtime, alloc *m
 }
 
 func (p *AstProgram) Start(ctx context.Context, alloc *memory.Allocator) (flux.Query, error) {
+	if p.opts == nil {
```

Comment:
I think I have misled you on my advice on how to handle the `now` time stuff. Here's the best way to fix it I think:
- Start by rolling back all the change in this file so far
- Add the following lines below, before calling `p.Program.Start`
```
deps := execdeps.NewExecutionDependencies(alloc, &p.Now, p.Logger)
ctx = deps.Inject(ctx)
```

At first I thought that we would not want to duplicate code needlessly, but it turns out that the value of `Now` may change after `Eval` is called, because it may process an `option now = () => ...` statement. This change is propagated to `p.Now` in `getSpec`, above. So we want to overwrite the old value if it has changed.

Reply:
I followed your conversation with Joshua and this makes sense.